src/main/java/com/pnudev/springonlinedelivery/controllers/OrderController.java

```diff
@@ -16,7 +25,23 @@ public class OrderController {
     private final OrderService orderService;
 
     @GetMapping
-    public List<Order> getOrders() {
+    public List<OrderDto> getOrders() {
         return orderService.getOrders();
     }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public OrderDto createOrder(@Valid @RequestBody OrderDto orderDto) {
+        return orderService.createOrder(orderDto);
+    }
+
+    @PutMapping(path = "/{id}")
+    public OrderDto updateOrder(@PathVariable Long id, @RequestBody OrderUpdateDto orderUpdateDto){
+        return orderService.updateOrder(id, orderUpdateDto);
+    }
+
+    @PutMapping(path = "/remove")
```

Comment:
`@DeleteMapping(path = "/{id}")`. This way you can remove id from request body

Reply:
The idea was that when the order was delivered, it would not be deleted but become done. This is necessary so that the user can view previous orders

Reply:
The idea is not to delete orders, but to change their status to Done. This is necessary so that the user can view previous orders

Reply:
Then use `@PutMapping(path = "/{id})` to pass id in path and remove it from request body. 
Also it seems strange that you are passing status in request body. Do you expect to have several statuses that need to be set inside request body? 
Because for me the operation seems that way: grab id from path -> set hardcoded status in service (not spcified in request) -> save

Reply:
But then there will be ambiguous mapping, and I have no clue on this

Reply:
Anyway, it's resolved rn